ConfigSpace/hyperparameters.py

```diff
@@ -70,6 +70,10 @@ def __repr__(self):
     def is_legal(self, value):
         raise NotImplementedError()
 
+    @abstractmethod
+    def is_legal_vector(self, value):
```

Comment:
Sorry, I don't immediately see what `is_legal_vector` is doing. Could you please add a docstring?